tests/lib/rules/no-unnecessary-act.test.ts

```diff
@@ -0,0 +1,782 @@
+          column: 15,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 13,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 14,
+          column: 15,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 16,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 19,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 22,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 25,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 28,
+          column: 15,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 31,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 34,
+          column: 15,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 37,
+          column: 9,
+        },
+        {
+          messageId: 'noUnnecessaryActTestingLibraryUtil',
+          line: 40,
+          column: 15,
+        },
+      ],
+    },
+    {
+      code: `// case: RTL act wrapping empty callback
+      import { act } from '@testing-library/react'
+
+      test('invalid case', async () => {
+        await act(async () => {})
+        act(() => {})
+        await act(async function () {})
+        act(function () {})
+      })
+      `,
+      errors: [
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 5, column: 15 },
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 6, column: 9 },
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 7, column: 15 },
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 8, column: 9 },
+      ],
+    },
+    {
+      settings: {
+        'testing-library/utils-module': 'test-utils',
+      },
+      code: `// case: RTL act wrapping empty callback - require version
+      const { act } = require('@testing-library/react');
+
+      test('invalid case', async () => {
+        await act(async () => {})
+        act(() => {})
+        await act(async function () {})
+        act(function () {})
+      })
+      `,
+      errors: [
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 5, column: 15 },
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 6, column: 9 },
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 7, column: 15 },
+        { messageId: 'noUnnecessaryActEmptyFunction', line: 8, column: 9 },
+      ],
+    },
+
+    // cases for act related to React Test Utils
+    {
+      settings: {
+        'testing-library/utils-module': 'custom-testing-module',
```

Comment:
We usually set this setting to `"test-utils"` for testing Aggressive Reporting disabled.

It has been named differently here and this is not by chance. This is another of the issues I found. If I set this to `"test-utils"`, I would have 2 different imports ending with the same string: `"test-utils"` and `"react-dom/test-utils"`. This causes our import checks to work incorrectly, so I'll have to handle it in a separate PR.

Meanwhile, I'm using a different custom module here to avoid collision with `"react-dom/test-utils"`.